lib/streamlit/MediaFileManager.py

```diff
@@ -108,21 +114,32 @@ class MediaFileManager(object):
 
     def __init__(self):
         # Dict of file ID to MediaFile.
-        self._files_by_id = (
-            WeakValueDictionary()
-        )  # type: WeakValueDictionary[str, MediaFile]
+        self._files_by_id = dict()
 
         # Dict[session ID][coordinates] -> MediaFile.
         self._files_by_session_and_coord = collections.defaultdict(
             dict
         )  # type: DefaultDict[str, Dict[str, MediaFile]]
 
-        # Since _files_by_id is a weak dict, when a MediaFile is removed from
-        # _files_by_session_and_coord it automatically gets removed from
-        # _files_by_id.
+    def _del_expired_files(self):
+        LOGGER.debug("Deleting expired files...")
+
+        # Get a flat set of every file ID in the session ID map.
+        active_file_ids = set()
+
+        for sessID in self._files_by_session_and_coord.keys():
+            for coord, file_id in self._files_by_session_and_coord[sessID].items():
+                active_file_ids.add(file_id)
+
+        # Remove any file that is currently not in the file_ids set AND has
+        # an expired TTD.
+        for file_id, mf in list(self._files_by_id.items()):
```

Comment:
Nit: You don't need `list()`. Also why not merge the two ifs? You can "rephrase" the code in way that reads just like the comment:

```py
now = time.time()

for file_id, mf in self._files_by_id.items():
    file_is_expired = mf.ttd < now
    if file_id not in active_file_ids and file_is_expired:
        del self._files_by_id[file_id]
```

Reply:
If i don't use list I get an error about modifying the dictionary in place.